app/models/form_element.rb

```diff
@@ -1,3 +1,26 @@
 class FormElement < ActiveRecord::Base
   belongs_to :form
+
+  validates :name, :label, :data_type, presence: true
+
+  before_validation :set_name
+  before_validation :set_position, on: :create
+
+
+  private
+
+  def set_name
+    return if label.empty?
+
+    self.name = label.
+      strip.
+      gsub(/\s+/, ' ').
+      gsub(/\s/, '_').
+      downcase
```

Comment:
does this mean that the `name` field exposed in the master form creation UI always gets overriden?